docs/utils/threading/worker-pool.md

```diff
@@ -148,3 +148,129 @@ functionality in Thunder, which will be further described in a different documen
 ## Conclusions
 
 To sum up, the main idea is not to reinvent the wheel. When creating plugins, developers should keep in mind they are working on a large system in an embedded environment. Because of that from an architectural point of view, a different set of rules applies than when working on developing PC or even mobile applications. We all have to be aware of limitations like the low amount of memory available, the difficulty of keeping a system with dozens of plugins scalable, or even the necessity to use abstractions to achieve portability. These are the main reasons why it is essential to use functionalities that are already given, instead of making things suboptimally on your own.
+
+## Scheduler
+### Overview
+`Workerpool` has the ability to schedule `jobs` using the `Scheduler` class. This can be very helpful when you do not want to start a `job` immediately but at a specific time. To learn more about how to create a `job` read [Workerpool](../threading/worker-pool) section of documentation.
+
+ To schedule a `job` we use several methods for this:
+
+* `Schedule()`
+* `Reschedule()`
+* `Revoke()`
+
+Using these methods, we can schedule the performance of a task, change the date on which it will be performed or cancel it altogether. 
+### Scheduler methods
+
+#### Schedule
+`Schedule()` is used to plan when to perform the `job` we specified. Using this method is very simple. We specify the `job` and the date on which to start performing it in the appropriate format.
+!!! note
+    In case the given time would point to the past, the task will start executing immediately.
+
+```cpp
+void Schedule(const Core::Time& time, const Core::ProxyType<IDispatch>& job) override
+        {
+            if (time > Core::Time::Now()) {
+                ASSERT(job.IsValid() == true);
+                ASSERT(_timer.HasEntry(Timer(this, job)) == false);
+
+                _timer.Schedule(time, Timer(this, job));
+            }
+            else {
+                _threadPool.Submit(job, Core::infinite);
+            }
+        }
+```
+
+
+#### Reschedule
+`Reschedule()` method is used to change the execution date of an already scheduled `job`. It is used in the same way as the `Schedule()` method. It works as follows, first the scheduled `job` is attempted to be canceled and then the Schedule with the newly specified time is used.
```

Comment:
Maybe instead of "It is used in the same way as the `Schedule()` method", we could be more literal and say that it takes the same parameters, since it is probably what you meant 😃